lib/ansible/plugins/inventory/linode.py

```diff
@@ -125,18 +132,85 @@ def _filter_by_config(self, regions, types):
                 if instance.type.id in types
             ]
 
-    def _add_instances_to_groups(self):
+    def _get_instance_public_ip(self, instance):
```

Comment:
We could merge the two `_get_instance*_ip` functions with something like?

```
def _get_instance_by_ip(self, instance, is_private=False):
    """Returns the IPv4 address of the instance."""
    if instance.ipv4 and len(instance.ipv4) > 0:
        for ip in instance.ipv4:
            parsed_ip = ipaddress.ip_address(ip)

            if parsed_ip.is_private and is_private:
                return ip

            if parsed_ip.is_private and not is_private
                raise AnsibleError('Expected public IP not found')

            return parsed_ip

    raise AnsibleError('Instance IPv4 address is missing or empty')
```

Less code :rocket:

Reply:
How about....
```
  def _get_instance_ip(self, instance, private=False):
        """Returns the public or private IP address of the instance"""
        if instance.ipv4 and len(instance.ipv4) > 0:
            result_ip = None
            for ip in instance.ipv4:
                if not ipaddress.ip_address(ip).is_private and not private:
                    result_ip = ip
                elif ipaddress.ip_address(ip).is_private and private:
                    result_ip = ip
            if result_ip:
                return result_ip
        else:
            raise AnsibleError(
                'Instance IPv4 address is missing or empty'
            )
```

Reply:
Yes, great!